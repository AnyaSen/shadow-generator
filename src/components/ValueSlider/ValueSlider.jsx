import React from "react";

import Slider from "@material-ui/core/Slider";
import Tooltip from "@material-ui/core/Tooltip";

function ValueLabelComponent(props) {
  const { children, open, value } = props;

  return (
    <Tooltip open={open} enterTouchDelay={0} placement="top" title={value}>
      {children}
    </Tooltip>
  );
}

export default function ValueSlider({
  heading,
  value,
  handleChange,
  maxValue,
  minValue
}) {
  return (
    <div>
      <p>{heading}</p>

      <Slider
        ValueLabelComponent={ValueLabelComponent}
        value={value}
        onChange={handleChange}
        max={maxValue}
        min={minValue}
        step={0.01}
      />
    </div>
  );
}
