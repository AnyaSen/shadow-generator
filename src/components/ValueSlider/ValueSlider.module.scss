@import "../../assets/scss/breakpoints";

.slider {
  width: 90%;
  @include sm {
    width: 80%;
  }
}
