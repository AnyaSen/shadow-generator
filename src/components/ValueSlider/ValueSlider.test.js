import React from "react";
import { render, cleanup } from "@testing-library/react";

import ValueSlider from "./ValueSlider";

describe("<ValueSlider />", () => {
  afterEach(cleanup);

  test("Should have correct text and have 2 children.", () => {
    const headerText = "height";

    const { getByText, getByTestId, container } = render(
      <ValueSlider heading={headerText} />
    );

    const header = getByText(`${headerText}`);

    const sliderContainer = getByTestId(`Slider`);

    expect(header).toBeTruthy();
    expect(sliderContainer.children.length).toEqual(2);

    expect(container.firstChild).toMatchSnapshot();
  });
});
