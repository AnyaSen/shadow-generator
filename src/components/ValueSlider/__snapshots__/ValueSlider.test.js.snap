// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValueSlider /> Should have correct text and have 2 children. 1`] = `
<div
  data-testid="Slider"
>
  <p>
    height
  </p>
  <span
    class="MuiSlider-root MuiSlider-colorPrimary"
  >
    <span
      class="MuiSlider-rail"
    />
    <span
      class="MuiSlider-track"
      style="left: 0%;"
    />
    <input
      type="hidden"
      value="NaN"
    />
    <span
      aria-orientation="horizontal"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="NaN"
      class="MuiSlider-thumb MuiSlider-thumbColorPrimary"
      data-index="0"
      role="slider"
      tabindex="0"
    />
  </span>
</div>
`;
